core/controlplane/config/encrypted_assets.go

```diff
@@ -558,22 +564,28 @@ func ReadOrEncryptAssets(dirname string, manageCertificates bool, caKeyRequiredO
 		if file.readEncrypted {
 			data, err := encryptor.EncryptedCredentialFromPath(path, file.defaultValue)
 			if err != nil {
-				return nil, fmt.Errorf("Error encrypting %s: %v", path, err)
+				return nil, fmt.Errorf("error encrypting %s: %v", path, err)
 			}
 
 			*file.data = *data
 			if err := data.Persist(); err != nil {
-				return nil, fmt.Errorf("Error persisting %s: %v", path, err)
+				return nil, fmt.Errorf("error persisting %s: %v", path, err)
 			}
 		} else {
 			raw, err := RawCredentialFileFromPath(path, file.defaultValue)
 			if err != nil {
-				return nil, fmt.Errorf("Error reading credential file %s: %v", path, err)
+				return nil, fmt.Errorf("error reading credential file %s: %v", path, err)
 			}
 			if file.expiryCheck {
-				if err := tlsutil.CheckAllCertsValid(path, raw.content); err != nil {
+				certs, err := tlsutil.ToCertificates(raw.content)
+				if err != nil {
 					return nil, err
 				}
+				for _, cert := range certs {
+					if cert.IsExpired() {
+						return nil, fmt.Errorf("the following certificate in file %s has expired:-\n\n%s", path, cert)
```

Comment:
Seems like we miss the useful info originally included in the error in [`CheckAllCertsValid`](https://github.com/kubernetes-incubator/kube-aws/pull/1299/files#diff-ff7418bed798ff0b9347acef6765cb23L192)?

Reply:
certificate has `String()` method that prints all the info about certificate. This method is used in show certificates command as well

Reply:
Ah! Awesome. I missed that. LGTM 👍